aggie-coding-club/Rev-Registration
# Added favicon + navbar logo

## Description

This adds a navbar logo and favicon to the site.

### Favicon

So first, most of the files were generated using http://realfavicongenerator.net , and I used the "favicon checker" on the home page to check that the favicons for different locations were working correctly.

You can use the favicon checker [here](https://realfavicongenerator.net/favicon_checker?protocol=http&site=www.revregistration.com) to see what the other things they test for are, but here's what we failed:

|iOS Safari Touch Icon|Classic, desktop browsers|
|--|--|
|![chrome_Dz4gcZ8muR](https://user-images.githubusercontent.com/7295783/96321344-afb39080-0fda-11eb-9f82-fb0f30ac7f34.png)|![chrome_qzslTii6ig](https://user-images.githubusercontent.com/7295783/96321349-b3dfae00-0fda-11eb-8a15-62f0b5bbda8f.png)|

I don't really care about the iOS one, since first we're not supporting mobile atm, and second I can't name a single time I've ever done the Safari website -> app shortcut thing in my 8 years of having an iPhone (pretty sure it's an OG feature). The second one also probably doesn't matter, since we're using the SVG favicon.

Second, the "main" favicon, `favicon.svg`, uses the CSS media property thing to detect the OS theme and switches from maroon to white when it detects dark mode. Note that you have to refresh the page to see this change (from what I can tell at least).

### NavBar logo

So I had to do a few things to get this to work both locally and on GCP. Because our `index.html` is served by django, it can dynamically determine what the static URL is to changed between `/static/` (for localhost) and `http://storage.googleapis.com/revregistration1.appspot.com/` (For GCP). To bring this functionality to the frontend, I used two things:

1. webpack's `DefinePlugin`, which functions in a really weird way - at compile time, it **replaces** the given word (`STATIC_URL` in this case) to whatever you have it defined as (excluding the declaration, but that isn't compiled to JS anyways). 
  a. That means that when you compile the project, and search for `STATIC_URL` in `main.js`, you won't see it anywhere 
  b. I added the `global.ts` file just so that we can treat `STATIC_URL` as a string without having webpack yell at us for not defining it.
2. the npm package `cross-env` to add the env var `PRODUCTION` whenever we run `npm run build`. This way it automatically changes from /static/ to the GCP URL whenever we run `npm run build`. 

## How to test

Build + run the site, and check that the logo on the navbar + favicon appear correct. Then to check the dark-mode SVG favicon, change your OS's theme and refresh.

## Screenshots

|Light mode|dark mode|
|--|--|
|![chrome_i2Gt6wy4QV](https://user-images.githubusercontent.com/7295783/96311201-96a4e280-0fce-11eb-8742-e51367cbcd0a.png)|![chrome_doKVWf6sDg](https://user-images.githubusercontent.com/7295783/96311212-9c9ac380-0fce-11eb-832f-7f78932f2509.png)|

Here's what the logo looks like on the scheduling page
![chrome_rz3tBhJM9I](https://user-images.githubusercontent.com/7295783/96311235-a9b7b280-0fce-11eb-8516-d5b199c74fdd.png)

## Related tasks

Closes #333
